core/tests/protractor_desktop/ExplorationTranslationTab.js

```diff
@@ -124,18 +124,93 @@ describe('Exploration translation', function() {
     explorationEditorMainTab.setInteraction('Continue');
     responseEditor = explorationEditorMainTab.getResponseEditor('default');
     responseEditor.setDestination('final card', true, null);
-    // Setup a terminating state.
     explorationEditorMainTab.moveToState('final card');
     explorationEditorMainTab.setInteraction('EndExploration');
     explorationEditorMainTab.moveToState('first');
     explorationEditorPage.saveChanges();
-
     explorationEditorPage.navigateToTranslationTab();
+    explorationEditorTranslationTab.exitTutorial();
     explorationEditorTranslationTab.expectNumericalStatusToMatch(
       '(0/8)');
+
+    explorationEditorTranslationTab.openUploadAudioModal();
+    var audioToUpload = '../data/cafe.mp3';
+    var audioAbsolutePath = path.resolve(__dirname, audioToUpload);
+    explorationEditorTranslationTab.audioElem().sendKeys(audioAbsolutePath);
+    explorationEditorTranslationTab.saveUploadedAudio(true);
+    explorationEditorTranslationTab.expectNumericalStatusToMatch(
+      '(1/8)');
+    explorationEditorTranslationTab.openUploadAudioModal();
+
+    var audioToUpload = '../data/img.png';
+    var audioAbsolutePath = path.resolve(__dirname, audioToUpload);
+    explorationEditorTranslationTab.audioElem().sendKeys(audioAbsolutePath);
+    expect(explorationEditorTranslationTab.errorMessage === undefined).
+      toBe(false);
+    expect(explorationEditorTranslationTab.saveUploadedAudioButton.
+      getAttribute('disabled')).toBe('true');
+    explorationEditorTranslationTab.expectNumericalStatusToMatch(
+      '(1/8)');
+    audioToUpload = '../data/cafe-over-five-minutes.mp3';
+    audioAbsolutePath = path.resolve(__dirname, audioToUpload);
+    explorationEditorTranslationTab.audioElem().sendKeys(audioAbsolutePath);
+    expect(explorationEditorTranslationTab.
+      errorMessageElement() === undefined).toBe(false);
+    explorationEditorTranslationTab.saveUploadedAudio();
+    expect(explorationEditorTranslationTab.saveUploadedAudioButton.
+      getAttribute('disabled')).toBe('true');
+    explorationEditorTranslationTab.expectNumericalStatusToMatch(
+      '(1/8)');
+
     users.logout();
   });
 
+  it('should provide correct status color for each state in the graph view',
+    function() {
+      var ALL_AUDIO_AVAILABLE_COLOR = 'rgb(22, 167, 101)';
+      var FEW_AUDIO_AVAILABLE_COLOR = 'rgb(233, 179, 48)';
+      var NO_AUDIO_AVAILABLE_COLOR = 'rgb(209, 72, 54)';
+
+      users.createUser('user@correctstatus.com', 'correctStatus');
+      users.login('user@correctstatus.com');
+      workflow.createExploration();
```

Comment:
Would it be possible to create a common exploration for both the tests, and run the tests on it? This can be done using a beforeAll() block, to do all common setup.

Reply:
This might take some time. Will get back to you on this ASAP. :)

Reply:
done